components/scream/src/share/atmosphere_process.hpp

```diff
@@ -35,6 +35,9 @@ enum class AtmosphereProcessType {
 class AtmosphereProcess
 {
 public:
+  using device_type      = DefaultDevice; // may need to template class on this
```

Comment:
I think we need to plan for a future in which kernels run on multiple device types. Certainly OpenMP on host and Cuda on device is possible in the near term.

Reply:
That's reasonable. Which means that, given that we templated the field structure on the device type, virtually every class in the driver infrastructure has to be templated on the device. I was trying to avoid this, to keep the code lighter (and break compilation into multiple TUs). But it seems we can't avoid that while maintaining flexibility (except maybe with some ETI).

Reply:
Yes, ETI for sure. None of the driver code will run on GPU, so it call be ETI'ed, regardless of Cuda or non-Cuda build (unlike the physics params, which if ETIed in Cuda build would then require RDC).